app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/DLCStatus.scala

```diff
@@ -1,360 +1,676 @@
+          contractId,
+          oracleInfo,
+          ContractInfo.fromTLV(contractInfoTLV),
+          DLCTimeouts(contractMaturity, contractTimeout),
+          feeRate,
+          totalCollateral,
+          localCollateral,
+          fundingTxId,
+          closingTxId,
+          oracleSigs.head,
+          outcome
+        )
+      case DLCState.Refunded =>
+        Refunded(
+          paramHash,
+          isInitiator,
+          tempContractId,
+          contractId,
+          oracleInfo,
+          ContractInfo.fromTLV(contractInfoTLV),
+          DLCTimeouts(contractMaturity, contractTimeout),
+          feeRate,
+          totalCollateral,
+          localCollateral,
+          fundingTxId,
+          closingTxId
+        )
+
     }
   }
 
-  /** The state where the remote party has successfully
-    * executed a unilateral execution by broadcasting both
-    * a CET and successfully spending their to_local output.
-    */
-  case class RemoteClaimed(
-      eventId: Sha256DigestBE,
+  def calculateOutcomeAndSig(
       isInitiator: Boolean,
       offer: DLCOffer,
       accept: DLCAccept,
       sign: DLCSign,
-      fundingTx: Transaction,
-      cet: Transaction,
-      closingTx: Transaction)
-      extends AcceptedDLCStatus {
-    override val statusString: String = "REMOTE CLAIMED"
-  }
+      cet: Transaction): (SchnorrDigitalSignature, DLCOutcomeType) = {
+    val cetSigs = cet
+      .asInstanceOf[WitnessTransaction]
+      .witness
+      .head
+      .asInstanceOf[P2WSHWitnessV0]
+      .signatures
+
+    require(cetSigs.size == 2,
+            s"There must be only 2 signatures, got ${cetSigs.size}")
+
+    val oraclePubKey = offer.oracleInfo.pubKey
+    val rVals = offer.oracleInfo.nonces
+
+    def aggregateR(numSigs: Int): SchnorrNonce = {
+      rVals.take(numSigs).map(_.publicKey).reduce(_.add(_)).schnorrNonce
+    }
 
-  /** The state where the remote party has published a
-    * CET but failed to spend their to_local output
-    * and this party has successfully spent that to_local
-    * output with a penalty transaction.
-    */
-  case class RemotePenalized(
-      eventId: Sha256DigestBE,
-      isInitiator: Boolean,
-      offer: DLCOffer,
-      accept: DLCAccept,
-      sign: DLCSign,
-      fundingTx: Transaction,
-      cet: Transaction,
-      penaltyTx: Transaction)
-      extends AcceptedDLCStatus {
-    override val statusString: String = "REMOTE PENALIZED"
-  }
+    def sigFromMsgAndSigs(
+        outcome: DLCOutcomeType,
+        adaptorSig: ECAdaptorSignature,
+        cetSig: ECDigitalSignature): SchnorrDigitalSignature = {
+      val (sigPubKey, numSigs) = outcome match {
+        case EnumOutcome(outcome) =>
+          val sigPoint = oraclePubKey.computeSigPoint(
+            CryptoUtil.sha256(outcome).bytes,
+            aggregateR(1))
+
+          (sigPoint, 1)
+        case UnsignedNumericOutcome(digits) =>
+          val sigPoint = digits
+            .zip(rVals.take(digits.length))
```

Comment:
So this seems a little suspicious to me 

1. Is it ok if we more `rVals` than digits? 
2. I posted earlier with concerns about ordering of `digits` and `rVals` How do we know the ordering is correct here?

Reply:
1. It is ok and expected, many CETs don't look at all digits
2. added a comment to `nonces` in response to earlier comment saying correct order is required